packages/terra-pills/src/private/_Pill.jsx

```diff
@@ -0,0 +1,301 @@
+/* eslint-disable jsx-a11y/role-supports-aria-props */ // lp052179
+import React, {
+  useCallback, useLayoutEffect, useRef, useState,
+} from 'react';
+import {
+  KEY_SPACE,
+  KEY_RETURN,
+  KEY_DELETE,
+  KEY_BACK_SPACE,
+} from 'keycode-js';
+import { injectIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import classNamesBind from 'classnames/bind';
+import Popup from 'terra-popup';
+import ResizeObserver from 'resize-observer-polyfill';
+import ThemeContext from 'terra-theme-context';
+import styles from './Pill.module.scss';
+
+const cx = classNamesBind.bind(styles);
+
+const propTypes = {
+  /**
+    * The html 'id' attribute for the pill, needed for proper keyboard interactions. (Required & must be unique)
+    */
+  id: PropTypes.string.isRequired,
+  /**
+   * The label text for the pill. (Required)
+   */
+  label: PropTypes.string.isRequired,
+  /**
+   * Secondary label text for the pill that displays a filter category
+   */
+  labelCategory: PropTypes.string,
+  /**
+   * Object returned along with the 'pillKey' in the `onRemove`
+   */
+  metaData: PropTypes.object,
+  /**
+   * To identify which pill is being removed.
+   */
+  pillKey: PropTypes.string,
```

Comment:
I'm not against having this prop but I wonder if it is truly needed. Are we able to just use the required `id` prop to send back in `onRemove` to indicate which pill is being removed?

If we do want to keep this, should it be required since it is needed to be returned in `onRemove`?

Reply:
I have the same concern - that pillKey is a redundant ID. What use case does pillKey fill that ID does not?

Reply:
removed pillKey and used ID now  ae07eb46c1332427df41f2b68be6f9007c75930c